### ISSUE TITLE
Method valid? overrides output to nil

### ISSUE THREAD

**Issue by unrooty** [Freshly/flow#97]
I have such State: 
```ruby
class PurchaseState < ApplicationState
  option :date
  option :total_amount
  option :number_of_items
  option :comment

  validates :total_amount, :number_of_items, numericality: { greater_than_or_equal_to: 0 }, allow_nil: true

  output :purchase
end
```
and such Operation:

```ruby
class Purchase::Create < ApplicationOperation
  def behavior
    state.purchase = Purchase.create!(date: state.date, total_amount: state.total_amount,
                                      comment: state.comment, number_of_items: state.number_of_items)
  end
end
```

Then I have following code: 
```ruby
result = Purchase::CreateFlow.trigger(date: Date.current)
result.state.purchase # #<Purchase id: 25, ... >
result.state.valid? # true
result.state.purchase # return nil right there
```
I really can't understand. Is it normal behavior that after valid? method call output returns nil or not?

**Comment by garside** [comment]
Hmm, confirmed this looks like a defect. I believe there's something wrong in the way the output interacts with the validation. 

Basically what's happening is that `result.state.valid?` is re-running the validations which is likely resetting the output.

**Comment by garside** [comment]
Confirmed the bug and added a fix! Will likely be part of the 1.0 release which I'm targeting tomorrow.

Thanks for your report @unrooty

**Comment by unrooty** [comment]
@garside thank you for quick fix! :)
